maas/data_source_maas_boot_resource.go

```diff
@@ -0,0 +1,108 @@
+package maas
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceMaasBootResources() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceMaasBootResourcesRead,
+		Description: "Provides a data source to manage MAAS bootresources.",
+
+		Schema: map[string]*schema.Schema{
+			"boot_resources": {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Description: "The set of boot resources for this os/release",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"architecture": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The architecture of this resource.",
+						},
+						"last_deployed": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The time of last deploy for this resource",
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The name of this resource.",
+						},
+						"subarches": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The subarches for this resource.",
+						},
+						"type": {
```

Comment:
Type can be either synced or uploaded (custom). Since we only care for synced, what about not keeping track of it with Terraform at all? This one requires changes to couple of places (docs included) but it can simplify the resources. In case we want to also support custom, we need to keep it but update our logic. I favor the synced only option but it's up to you